src/Components/RosTable/RosTable.js

```diff
@@ -0,0 +1,130 @@
+    let applyClasses = 'recommendations';
+    if (textValue === 0) {
+        applyClasses += ' green-400';
+    }
+
+    return actionLink(id, textValue, applyClasses, '#');
+};
+
+class RosTable extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            columns: [
+                { title: 'System Name', cellFormatters: [expandable] },
+                { title: 'CPU score' },
+                { title: 'Memory score' },
+                { title: 'I/O score' },
+                { title: 'Recommendations' },
+                { title: 'State' }
+            ],
+            rows: this.createRows()
+        };
+        this.onCollapse = this.onCollapse.bind(this);
+    }
+
+    onCollapse(event, rowKey, isOpen) {
+        const { rows } = this.state;
+        /**
+         * Please do not use rowKey as row index for more complex tables.
+         * Rather use some kind of identifier like ID passed with each row.
+         */
+        rows[rowKey].isOpen = isOpen;
+        this.setState({
+            rows
+        });
+    }
+
+    createRows() {
+        const rowsData = this.props.systems;
+
+        if (rowsData.length !== 0) {
+            return flatMap(rowsData, (row, index) => {
+                const { cpu_score: cpuScore, memory_score: memoryScore, io_score: IOScore } = row.performance_score;
+                const { cloud_provider: cloudProvider, instance_type: instanceType,
+                    idling_time: idlingTime, io_wait: ioWait } = row.facts;
+                const { id, recommendation_count: recommendationCount } = row;
+
+                return [
+                    {
+                        id: index,
+                        isOpen: false,
+                        cells: [
+                            { title: renderSystemLink(id, row.display_name) },
+                            { title: <ProgressScoreBar measureLocation='outside' valueScore={cpuScore} /> },
+                            { title: <ProgressScoreBar measureLocation='outside' valueScore={memoryScore} /> },
+                            { title: <ProgressScoreBar measureLocation='outside' valueScore={IOScore} /> },
+                            { title: renderRecommendations(id, recommendationCount) },
+                            { title: row.state }
+
+                        ]
+                    },
+                    {
+                        cells: [
+                            {
+                                title: <ExpandedRow {
+                                    ...{ id, cloudProvider, instanceType, idlingTime, ioWait }
+                                } />
+                            }
+                        ],
+                        parent: index * 2
+                    }
+                ];
+            });
+        } else {
+            let emptyRow = <EmptyTable>
+                <EmptyStateDisplay title={ 'No matching records found' } />
```

Comment:
Direct string, no need to wrap in `{}`
```suggestion
                <EmptyStateDisplay title="No matching records found" />
```